### PR TITLE
Nextflow conversion

### PR DESCRIPTION
**Pull request by hyunhwan-bcm**
The PR includes numerous changes and should be carefully reviewed. @sasidharPasupuleti and @ZaahidShaik, please try to run this PR and verify its functionality. It would be beneficial to test it against a previously completed run.

## The PR Summary
- [x] convert the pipeline to a nextflow version
- [x] add error handling strategy 
- [ ] output test 
  - [x] on demo data (demo.vcf.gz, 8.7M)
  - [x] on real data (6656513962b9002680181ed0.recode.vcf.gz, 930K)</strike>
- [ ] integrity with README
- [ ] code review
- [ ] code optimization 

## How to use (or test)

Here is  an example:

```
nextflow run main.nf --run_id 12345 --ref_ver "hg19" --outdir testout_061024 \
                                    --input_hpo /home/hwan/AI_MARRVEL/test_data/hpos.txt \
                                    --input_vcf /home/hwan/AI_MARRVEL/test_data/demo.vcf.gz \
                                    --ref_dir /home/hwan/aim_data_dependencies_04_24_2024_sasi
```

You can also change the parameters in `nextflow.config` - you can run `nextflow run main.nf`

### CODE REVIEW

File: scripts/proc.sh
Comment (jylee-bcm) [inline scripts/proc.sh:1]: If it's not used anymore, let's remove it

File: nextflow.config
Comment (jylee-bcm) [inline nextflow.config:1]: * Rename `nextflow.config` to `nextflow.config.example`
* Add `nextflow.config` to `.gitignore` list
* `nextflow.config.example `contain an instruction saying: Use this example file to create nextflow.config and change the values for: ref_dir, input_vcf, input_hpo`

File: .gitignore
Comment (jylee-bcm) [inline .gitignore:1]: Add nextflow.config

**Review by jylee-bcm — CHANGES_REQUESTED** [review]
I think `nextflow.config` should contain different values for each environment, so I'd like to include it into `.gitignore`. For the details, please read the comment on related code.

**Comment by hyunhwan-bcm** [comment]
@jylee-bcm please review my latest commit to this branch.

**Review by jylee-bcm — COMMENTED** [review]
I think it's better to remove /run/ directory, so that git can track the difference.
I will upload PR for it.

**Comment by hyunhwan-bcm** [comment]
I thought I remove them, but it wasn't pushed. I am fine that if you delete the directory.

File: utils/run_diff.py
Comment (jylee-bcm) [inline utils/run_diff.py:1]: Is it actively used? if not, remove it

**Comment by jylee-bcm** [comment]
> I thought I remove them, but it wasn't pushed. I am fine that if you delete the directory.
You were right, just git couldn't track the difference.

**Comment by hyunhwan-bcm** [comment]
You can remove it just for comparison of every column of dataframes

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Review by jylee-bcm — APPROVED** [review]
LGTM

**Pull request merged**